IllegalArgumentException: port p must be in range 0 <= p <= 65535 when using client-endpoint-url without explicit port

```
2024-08-06 08:49:17,422 WARN  [org.apa.cxf.pha.PhaseInterceptorChain] (executor-thread-1) Interceptor for {http://vimis.rosminzdrav.ru/}clinrec#{http://vimis.rosminzdrav.ru/}clinrecList has thrown exception, unwinding now>         at io.vertx.core.impl.Arguments.requireInRange(Arguments.java:44)
         at io.vertx.core.net.SocketAddress.inetSocketAddress(SocketAddress.java:64)
         at io.vertx.core.http.impl.HttpClientImpl.doRequest(HttpClientImpl.java:266)
         at io.vertx.core.http.impl.HttpClientImpl.request(HttpClientImpl.java:191)
         at io.quarkiverse.cxf.vertx.http.client.VertxHttpClientHTTPConduit$RequestBodyHandler.handle(VertxHttpClientHTTPConduit.java:419)
         at io.quarkiverse.cxf.vertx.http.client.VertxHttpClientHTTPConduit$RequestBodyHandler.handle(VertxHttpClientHTTPConduit.java:343)
         at io.quarkiverse.cxf.vertx.http.client.VertxHttpClientHTTPConduit$RequestBodyOutputStream.write(VertxHttpClientHTTPConduit.java:305)
         at org.apache.cxf.io.CacheAndWriteOutputStream.write(CacheAndWriteOutputStream.java:81)
         at org.apache.cxf.io.AbstractWrappedOutputStream.write(AbstractWrappedOutputStream.java:51)
         at com.ctc.wstx.io.UTF8Writer.flush(UTF8Writer.java:100)
         at com.ctc.wstx.sw.BufferingXmlWriter.flush(BufferingXmlWriter.java:242)
         at com.ctc.wstx.sw.BaseStreamWriter.flush(BaseStreamWriter.java:260)
         at org.apache.cxf.binding.soap.saaj.SAAJOutInterceptor$SAAJOutEndingInterceptor.handleMessage(SAAJOutInterceptor.java:214)
         at org.apache.cxf.binding.soap.saaj.SAAJOutInterceptor$SAAJOutEndingInterceptor.handleMessage(SAAJOutInterceptor.java:173)
         at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:307)
         at org.apache.cxf.endpoint.ClientImpl.doInvoke(ClientImpl.java:530)
         at org.apache.cxf.endpoint.ClientImpl.invoke(ClientImpl.java:441)
         at org.apache.cxf.endpoint.ClientImpl.invoke(ClientImpl.java:356)
         at org.apache.cxf.endpoint.ClientImpl.invoke(ClientImpl.java:314)
         at org.apache.cxf.frontend.ClientProxy.invokeSync(ClientProxy.java:96)
         at org.apache.cxf.jaxws.JaxWsClientProxy.invoke(JaxWsClientProxy.java:140)
         at jdk.proxy2/jdk.proxy2.$Proxy58.clinrecList(Unknown Source)
         at su.medsoft.rir.vimis.contract.ClinRecClient.getClinicalRecommendationList(ClinRecClient.java:21)
         at su.medsoft.rir.vimis.contract.ClinRecClient_ClientProxy.getClinicalRecommendationList(Unknown Source)
         at su.medsoft.rir.vimis.ClinicalRecommendationService.getRecommendationList(ClinicalRecommendationService.java:16)
         at su.medsoft.rir.vimis.ClinicalRecommendationService_ClientProxy.getRecommendationList(Unknown Source)
         at su.medsoft.rir.vimis.ClinicalRecommendationResource.getList(ClinicalRecommendationResource.java:27)
         at su.medsoft.rir.vimis.ClinicalRecommendationResource$quarkusrestinvoker$getList_a69a6af112009490ce87e7505a1dd8f585aa8958.invoke(Unknown Source)
         at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
         at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
         at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
         at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:635)
         at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
         at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
         at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
         at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
         at java.base/java.lang.Thread.run(Thread.java:1583)
 2024-08-06 08:49:17,427 ERROR [su.med.rir.vim.RestResponseExceptionHandler] (executor-thread-1) [id=2e6b7111-8b80-4eba-877a-4013abfaf558] Произошла неизвестная ошибка: jakarta.xml.ws.soap.SOAPFaultException: port p must b>
         at org.apache.cxf.jaxws.JaxWsClientProxy.mapException(JaxWsClientProxy.java:195)
         at org.apache.cxf.jaxws.JaxWsClientProxy.invoke(JaxWsClientProxy.java:145)
         at jdk.proxy2/jdk.proxy2.$Proxy58.clinrecList(Unknown Source)
         at su.medsoft.rir.vimis.contract.ClinRecClient.getClinicalRecommendationList(ClinRecClient.java:21)
         at su.medsoft.rir.vimis.contract.ClinRecClient_ClientProxy.getClinicalRecommendationList(Unknown Source)
         at su.medsoft.rir.vimis.ClinicalRecommendationService.getRecommendationList(ClinicalRecommendationService.java:16)
         at su.medsoft.rir.vimis.ClinicalRecommendationService_ClientProxy.getRecommendationList(Unknown Source)
         at su.medsoft.rir.vimis.ClinicalRecommendationResource.getList(ClinicalRecommendationResource.java:27)
         at su.medsoft.rir.vimis.ClinicalRecommendationResource$quarkusrestinvoker$getList_a69a6af112009490ce87e7505a1dd8f585aa8958.invoke(Unknown Source)
         at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
         at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
         at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
         at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:635)
         at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
         at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
         at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
         at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
         at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
         at java.base/java.lang.Thread.run(Thread.java:1583)
 Caused by: java.lang.IllegalArgumentException: port p must be in range 0 <= p <= 65535
         at io.vertx.core.impl.Arguments.requireInRange(Arguments.java:44)
         at io.vertx.core.net.SocketAddress.inetSocketAddress(SocketAddress.java:64)
         at io.vertx.core.http.impl.HttpClientImpl.doRequest(HttpClientImpl.java:266)
         at io.vertx.core.http.impl.HttpClientImpl.request(HttpClientImpl.java:191)
         at io.quarkiverse.cxf.vertx.http.client.VertxHttpClientHTTPConduit$RequestBodyHandler.handle(VertxHttpClientHTTPConduit.java:419)
         at io.quarkiverse.cxf.vertx.http.client.VertxHttpClientHTTPConduit$RequestBodyHandler.handle(VertxHttpClientHTTPConduit.java:343)
         at io.quarkiverse.cxf.vertx.http.client.VertxHttpClientHTTPConduit$RequestBodyOutputStream.write(VertxHttpClientHTTPConduit.java:305)
         at org.apache.cxf.io.CacheAndWriteOutputStream.write(CacheAndWriteOutputStream.java:81)
         at org.apache.cxf.io.AbstractWrappedOutputStream.write(AbstractWrappedOutputStream.java:51)
         at com.ctc.wstx.io.UTF8Writer.flush(UTF8Writer.java:100)
         at com.ctc.wstx.sw.BufferingXmlWriter.flush(BufferingXmlWriter.java:242)
         at com.ctc.wstx.sw.BaseStreamWriter.flush(BaseStreamWriter.java:260)
         at org.apache.cxf.binding.soap.saaj.SAAJOutInterceptor$SAAJOutEndingInterceptor.handleMessage(SAAJOutInterceptor.java:214)
         at org.apache.cxf.binding.soap.saaj.SAAJOutInterceptor$SAAJOutEndingInterceptor.handleMessage(SAAJOutInterceptor.java:173)
         at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:307)
         at org.apache.cxf.endpoint.ClientImpl.doInvoke(ClientImpl.java:530)
         at org.apache.cxf.endpoint.ClientImpl.invoke(ClientImpl.java:441)
         at org.apache.cxf.endpoint.ClientImpl.invoke(ClientImpl.java:356)
         at org.apache.cxf.endpoint.ClientImpl.invoke(ClientImpl.java:314)
         at org.apache.cxf.frontend.ClientProxy.invokeSync(ClientProxy.java:96)
         at org.apache.cxf.jaxws.JaxWsClientProxy.invoke(JaxWsClientProxy.java:140)
         ... 18 more
 2024-08-06 08:49:17,487 INFO  [io.qua.htt.access-log] (executor-thread-1) 192.168.17.120 - - [06/Aug/2024:08:49:17 +0300] "GET /1.0/vimis/ssz/clinrec/list HTTP/1.1" 500 121 1168
```

This error above started to occur, when i've added quarkus.cxf.http-conduit-factory = VertxHttpClientHTTPConduitFactory

this is my config:
```
quarkus.cxf.client.clinrec.wsdl=wsdl/clinrec.wsdl
quarkus.cxf.client.clinrec.client-endpoint-url = http://ips.rosminzdrav.ru/6f4cd9cc5564d
quarkus.cxf.client.clinrec.service-interface = su.medsoft.rir.vimis.contract.clinrec.Clinrec
quarkus.cxf.client.clinrec.handlers=su.medsoft.rir.vimis.contract.handlers.SigningSoapHandler,su.medsoft.rir.vimis.contract.handlers.TransportSoapHeader
```
After that error no logs (despite quarkus.cxf.logging.enabled-for=both), which seems client not working.
Quarkus 3.13.0, Java 21

--- a/docs/antora.yml
+++ b/docs/antora.yml
@@ -8,7 +8,7 @@ asciidoc:
   attributes:
 
     # Versions
-    quarkus-version: 3.13.0 # replace ${quarkus.version}
+    quarkus-version: 3.13.1 # replace ${quarkus.version}
     quarkus-cxf-version: 3.13.1 # replace ${release.current-version}
 
     # Toggle whether some page elements are rendered

--- a/extensions/core/vertx-client/src/main/java/io/quarkiverse/cxf/vertx/http/client/VertxHttpClientHTTPConduit.java
+++ b/extensions/core/vertx-client/src/main/java/io/quarkiverse/cxf/vertx/http/client/VertxHttpClientHTTPConduit.java
@@ -160,10 +160,16 @@ public class VertxHttpClientHTTPConduit extends HTTPConduit {
                 : uri.getPath();
         requestOptions
                 .setMethod(method)
-                .setPort(uri.getPort())
                 .setHost(uri.getHost())
                 .setURI(pathAndQuery)
                 .setConnectTimeout(determineConnectionTimeout(message, csPolicy));
+
+        final int port = uri.getPort();
+        if (port >= 0) {
+            requestOptions
+                    .setPort(uri.getPort());
+        }
+
         final RequestContext requestContext = new RequestContext(
                 uri,
                 requestOptions,

